cms/server/contest/handlers/tasksubmission.py

```diff
@@ -67,9 +67,16 @@ class SubmitHandler(ContestHandler):
     """
 
     @tornado_web.authenticated
-    @actual_phase_required(0, 3)
+    @actual_phase_required(0, 1, 2, 3, 4)
```

Comment:
Hmm, I think we should not allow phase 4 here?

```
    * +4: the user cannot compete because the contest has already
          stopped. analysis mode has already finished or has been
          disabled for this contest.
```

When analysis mode is finished, the actual contest is over and it does not make sense to still allow submissions at this point.